integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -268,6 +277,94 @@ fn fork_before_resharding_block(
     )
 }
 
+/// Returns a loop action that invokes a costly method from a contract multiple times per block height.
+fn call_burn_gas_contract(
+    signer_id: AccountId,
+    receiver_id: AccountId,
+) -> Box<dyn Fn(&[TestData], &mut TestLoopData, TestLoopDataHandle<ClientActorInner>)> {
+    const TX_CHECK_BLOCKS_AFTER_RESHARDING: u64 = 5;
+    const GAS_PER_CALL: Gas = 275 * TGAS;
+    const CALLS_PER_BLOCK_HEIGHT: u64 = 5;
+
+    let resharding_height = Cell::new(None);
+    let nonce = Cell::new(102);
+    let txs = Cell::new(vec![]);
+    let latest_height = Cell::new(0);
+    let signer: Signer = create_user_test_signer(&signer_id).into();
+
+    Box::new(
+        move |node_datas: &[TestData],
+              test_loop_data: &mut TestLoopData,
+              client_handle: TestLoopDataHandle<ClientActorInner>| {
+            let client_actor = &mut test_loop_data.get_mut(&client_handle);
+            let tip = client_actor.client.chain.head().unwrap();
+
+            // Run this action only once at every block height.
+            if latest_height.get() == tip.height {
+                return;
+            }
+            latest_height.set(tip.height);
+
+            // After resharding: wait some blocks and check that all txs have been executed correctly.
+            if let Some(height) = resharding_height.get() {
+                if tip.height > height + TX_CHECK_BLOCKS_AFTER_RESHARDING {
+                    for (tx, tx_height) in txs.take() {
+                        let tx_outcome =
+                            client_actor.client.chain.get_partial_transaction_result(&tx);
+                        let status = tx_outcome.as_ref().map(|o| o.status.clone());
+                        let status = status.unwrap();
+                        tracing::debug!(target: "test", ?tx_height, ?tx, ?status, "transaction status");
+                        assert_matches!(status, FinalExecutionStatus::SuccessValue(_));
+                    }
+                }
+            } else {
+                if next_block_has_new_shard_layout(client_actor.client.epoch_manager.clone(), &tip)
+                {
+                    tracing::debug!(target: "test", height=tip.height, "resharding height set");
+                    resharding_height.set(Some(tip.height));
+                }
```

Comment:
Can you also assert that there are delayed receipts at the post state of the resharding block?

Reply:
Good point, I've done a check using `congestion_info`.